### PR TITLE
Add annotation to configure healthcheck retries for EtcdadmCluster

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Description of changes:*
This PR adds an annotation `etcdcluster.cluster.x-k8s.io/healthcheck-paused` which when added to the `EtcdadmCluster`, pauses healthchecks for all the ETCD machines for that cluster.

It also makes the healthcheck interval configurable using the `--healthcheck-interval` flag, with the same default of 30 seconds. This also helps with unit tests since we can change this value so the unit tests don't have to wait 30 seconds just for a healthcheck loop.

Also added a unit test that verifies that the healthcheck is paused when the annotation is set on the `EtcdadmCluster` object.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: controllers/periodic_healthcheck_test.go [hunk 1]
```diff
@@ -15,6 +21,35 @@ import (
...
+		HealthCheckInterval: 1, // override the healthcheck interval to 1 second
+	}
+
+	done := make(chan struct{})
+
+	// Stop the healthcheck loop after 5 seconds
+	go func() {
+		time.Sleep(5 * time.Second)
+		close(done)
+	}()
```
Comment (g-gaston) [inline controllers/periodic_healthcheck_test.go:45]: instead you could use t.Cleanup
It's simpler and you don't have to deal with concurrency

File: api/v1beta1/etcdadmcluster_types.go [hunk 2]
```diff
@@ -26,6 +26,8 @@ import (
 const (
 	UpgradeInProgressAnnotation = "etcdcluster.cluster.x-k8s.io/upgrading"
 
+	PausedHealthCheckAnnotation = "etcdcluster.cluster.x-k8s.io/healthcheck-paused"
```
Comment (g-gaston) [inline api/v1beta1/etcdadmcluster_types.go:29]: What if we made this annotation something like `etcdcluster.cluster.x-k8s.io/healthcheck-retries`?
That would give users the option to configure it a little bit on top of being able to disable it. For example, a value of 0 could mean healthcheck disabled.

File: controllers/controller_test.go [hunk 3]
```diff
@@ -463,8 +463,24 @@ func newEtcdadmCluster(cluster *clusterv1.Cluster) *etcdv1.EtcdadmCluster {
...
+	return etcdCluster
 }
 
+type etcdadmClusterTestOpt func(e *etcdv1.EtcdadmCluster)
+
+func withPausedAnnotation(e *etcdv1.EtcdadmCluster) {
+	e.SetAnnotations(map[string]string{etcdv1.HealthCheckRetriesAnnotation: "0"})
+}
+
+// func withOwnerRef(e *etcdv1.EtcdadmCluster) {
```
Comment (ahreehong) [inline controllers/controller_test.go:480]: Can you remove the commented code here?

File: controllers/controller_test.go [hunk 4]
```diff
@@ -433,8 +433,8 @@ func newClusterWithExternalEtcd() *clusterv1.Cluster {
 	}
 }
 
-func newEtcdadmCluster(cluster *clusterv1.Cluster) *etcdv1.EtcdadmCluster {
-	return &etcdv1.EtcdadmCluster{
+func newEtcdadmCluster(cluster *clusterv1.Cluster, opts ...etcdadmClusterTestOpt) *etcdv1.EtcdadmCluster {
```
Comment (ahreehong) [inline controllers/controller_test.go:436]: Are these changes still needed?

File: controllers/periodic_healthcheck_test.go [hunk 5]
```diff
@@ -15,6 +21,35 @@ import (
...
+
+	r.startHealthCheckLoop(context.Background(), done)
+
+	g.Expect(recordedLogs.All()).To(Not(BeEmpty()))
+	g.Expect(recordedLogs.All()[recordedLogs.Len()-1].Message).To(Equal("HealthCheck paused for EtcdadmCluster, skipping"))
```
[the remark this answers is not in the document]
Comment (abhinavmpandey08) [reply controllers/periodic_healthcheck_test.go:89]: fixed

File: controllers/periodic_healthcheck_test.go (hunk 1 above)
Comment (abhinavmpandey08) [reply controllers/periodic_healthcheck_test.go:78]: removed

File: api/v1beta1/etcdadmcluster_types.go (hunk 2 above)
Comment (abhinavmpandey08) [reply api/v1beta1/etcdadmcluster_types.go]: done

File: controllers/controller_test.go (hunk 4 above)
Comment (abhinavmpandey08) [reply controllers/controller_test.go]: nope, removed

**Review by ahreehong — APPROVED** [review]
